packages/medusa/src/services/sales-channel-location.ts

```diff
@@ -102,6 +102,22 @@ class SalesChannelLocationService extends TransactionBaseService {
 
     return locations.map((l) => l.location_id)
   }
+
+  /**
+   * Lists the sales channels associated with a stock location.
+   * @param {string} salesChannelId - The ID of the stock location.
+   * @returns {Promise<string[]>} A promise that resolves with an array of sales channel IDs.
+   */
+  async listSalesChannelIds(locationId: string): Promise<string[]> {
+    const manager = this.transactionManager_ || this.manager_
+    const location = await this.stockLocationService.retrieve(locationId)
+
+    const salesChannelLocations = await manager.find(SalesChannelLocation, {
```

Comment:
suggestion: add in the config `select: ["sales_channel_id"]` as we don't need the rest of the columns